### PR TITLE
Add BMaaS to CRC devsetup

### PR DESCRIPTION
**Pull request by hjensas**
Add make targets in devsetup Makefile to configure libvirt networking, vm instances, CRC networking and a virtual RedFish emulator (sushy-emulator).

Access RedFish via:
  http://sushy-emulator.apps-crc.testing

Jira: OSP-23574

### CODE REVIEW

**Comment by hjensas** [comment]
recheck

File: devsetup/scripts/bmaas/network-setup.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# set -x
+
```
Comment (stuggi) [inline devsetup/scripts/bmaas/network-setup.sh:3]: how about we make this a generic network-setup.sh script for devsetup. right now we use the default libvirt network for attaching the edpm node and do a static dhcp reservations for crc and the edpm vms. this sometimes result in issues where when you re-create the VM and the new VM with a new mac gets a different IP address even if there is a static reservation. seems that dnsmasq still has a lease for the old one. the proper way could be like in this script to destroy and recreate the network and recreate it when the env get re-created, which we probably don't want to do with the default libvirt network.

Comment (hjensas) [reply devsetup/scripts/bmaas/network-setup.sh:3]: Sure, I'll move this script out of the bmaas subdirectory and do some slight modifications to make it more generic.

Comment (hjensas) [reply devsetup/scripts/bmaas/network-setup.sh]: @stuggi

I removed the "crc-bmaas" mentions in the script, moved it up to the /scripts directory.
Changed the default network name to crc-net-iso, and used 172.16.0.1/24 as the IP address.
I think that's all that is needed for generic use?

Comment (stuggi) [reply devsetup/scripts/bmaas/network-setup.sh]: yes, thanks! lgtm

**Comment by hjensas** [comment]
recheck

File: devsetup/scripts/bmaas/br-baremetal-bridge.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    temp_file=$(mktemp -p "$MY_TMP_DIR")
+    cat << EOF > "$temp_file"
+apiVersion: nmstate.io/v1
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: br-baremetal
+spec:
+  desiredState:
+    interfaces:
+    - name: br-baremetal
```
[the remark this answers is not in the document]
Comment (hjensas) [reply devsetup/scripts/bmaas/br-baremetal-bridge.sh]: I've updated the change to use $BMAAS_NETWORK_NAME for this.

**Comment by Sandeepyadav93** [comment]
+1 I tested this patch last week to create bmaas lab on libvirt end and it worked fine, I managed to successfully provision bmaas nodes.

**Review by rabi — APPROVED** [review]
/lgtm

**Pull request merged**